researchstack-sdk/src/main/java/org/sagebionetworks/bridge/researchstack/BridgeDataProvider.java

```diff
@@ -167,63 +181,90 @@ public boolean isConsented() {
+    public void saveConsent(Context context, @NonNull TaskResult consentResult) {
+        logger.warn("saveConsent called -- DEPRECATED");
+        giveConsent(bridgeConfig.getStudyId(), createConsentSignature(consentResult))
+                .toBlocking().value();
     }
 
     @Override
     public void saveLocalConsent(Context context, ConsentSignatureBody signatureBody) {
-        giveConsent(bridgeConfig.getStudyId(), createConsentSignature(signatureBody));
+        logger.debug("saveLocalConsent called for: " + signatureBody);
+        LocalDate birthdate = null;
+        if (signatureBody.birthdate != null) {
+            birthdate = LocalDate.fromDateFields(signatureBody.birthdate);
+        }
+        authenticationManager.giveConsentSync(bridgeConfig.getStudyId(),
+                signatureBody.name,
+                birthdate,
+                signatureBody.imageData,
+                signatureBody.imageMimeType,
+                toSharingScope(signatureBody.scope));
+    }
+
+    private SharingScope toSharingScope(String sharingScope) {
+        SharingScope scopeEnum = SharingScope.NO_SHARING;
+        for (SharingScope scope : SharingScope.values()) {
+            if (scope.toString().equals(sharingScope)) {
+                scopeEnum = scope;
+            }
+        }
+        return scopeEnum;
     }
 
     @Nullable
-    protected ConsentSignature createConsentSignature(ConsentSignatureBody consentSignatureBody) {
+    protected ConsentSignature createConsentSignature(@Nullable ConsentSignatureBody
+                                                              consentSignatureBody) {
         if (consentSignatureBody == null) {
             return null;
         }
         ConsentSignature signature = new ConsentSignature();
         signature.setName(consentSignatureBody.name);
-        signature.setBirthdate(LocalDate.fromDateFields(consentSignatureBody.birthdate));
+        if (consentSignatureBody.birthdate != null) {
+            signature.setBirthdate(LocalDate.fromDateFields(consentSignatureBody.birthdate));
+        }
+        SharingScope sharingScope = toSharingScope(consentSignatureBody.scope);
         signature.setImageData(consentSignatureBody.imageData);
         signature.setImageMimeType(consentSignatureBody.imageMimeType);
-        SharingScope sharingScope = SharingScope.NO_SHARING;
-        for (SharingScope scope : SharingScope.values()) {
-            if (scope.toString().equals(consentSignatureBody.scope)) {
-                sharingScope = scope;
-            }
-        }
         signature.setScope(sharingScope);
         return signature;
     }
 
     @Nullable
-    protected ConsentSignatureBody createConsentSignatureBody(ConsentSignature consentSignature) {
+    protected ConsentSignatureBody createConsentSignatureBody(@Nullable ConsentSignature
+                                                                      consentSignature) {
```

Comment:
Similarly, should we add unit tests for this? (Called by loadLocalConsent().) Again, we should have two test cases, one for all fields, one for only required fields.

Reply:
Wrote a test that covers this